libs/spandrel/spandrel/architectures/PLKSR/__init__.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+import math
+from typing import Union
+
+from typing_extensions import override
+
+from spandrel.util import KeyCondition, get_seq_len
+
+from ...__helpers.model_descriptor import (
+    Architecture,
+    ImageModelDescriptor,
+    SizeRequirements,
+    StateDict,
+)
+from .arch.PLKSR import PLKSR
+from .arch.RealPLKSR import RealPLKSR
+
+_PLKSR = Union[PLKSR, RealPLKSR]
+
+
+class PLKSRArch(Architecture[_PLKSR]):
+    def __init__(self) -> None:
+        super().__init__(
+            id="PLKSR",
+            detect=KeyCondition.has_all(
+                "feats.0.weight",
+                "feats.1.lk.conv.weight",
+                "feats.1.refine.weight",
+            )
+            and KeyCondition.has_any(
+                "feats.1.channe_mixer.0.weight",
+                "feats.1.channel_mixer.0.weight",
+            ),
+        )
+
+    @override
+    def load(self, state_dict: StateDict) -> ImageModelDescriptor[_PLKSR]:
+        scale: int
+        in_channels = 3
+        out_channels = 3
+        dim = 64
+        n_blocks = 28
+        scale = 4
+        kernel_size = 17
+        split_ratio = 0.25
+        use_ea = True
+
+        # RealPLKSR only
+        norm_groups = 4  # un-detectable
+        dropout = 0  # un-detectable
+
+        tags: list[str] = []
+
+        size_requirements = SizeRequirements(minimum=1)
+
+        if "feats.0.weight" in state_dict:
+            dim = state_dict["feats.0.weight"].shape[0]
+
+        total_feat_layers = get_seq_len(state_dict, "feats")
+
+        upscale_key = list(state_dict.keys())[-2]
+        if upscale_key in state_dict:
+            scale_shape = state_dict[upscale_key].shape[0]
+            scale = math.isqrt(scale_shape // out_channels)
+
+        if "feats.1.lk.conv.weight" in state_dict:
+            kernel_size = state_dict["feats.1.lk.conv.weight"].shape[2]
+            after_split = state_dict["feats.1.lk.conv.weight"].shape[0]
+            split_ratio = after_split / dim
+
+        if "feats.1.attn.f.0.weight" not in state_dict:
+            use_ea = False
+
+        # Yes, the normal version has this typo.
+        if "feats.1.channe_mixer.0.weight" in state_dict:
+            self._name = "PLKSR"

```

Comment:
TIL that `_variable_name` class fields aren't private but protected from the perspective of the type checker.

Yeah, this was not intended. `Architecture` instances are supposed to be immutable. So this is not allowed. I would suggest adding "Real" as a tag for `RealPLKSR`.